src/components/MainScheduler/index.jsx

```diff
@@ -0,0 +1,220 @@
+    },
+    {
+        title: 'Install New Database',
+        startDate: new Date(2022, 1, 17, 15, 45),
+        endDate: new Date(2022, 1, 18, 12, 15),
+    },
+    {
+        title: 'Approve New Online Marketing Strategy',
+        startDate: new Date(2022, 1, 18, 12, 35),
+        endDate: new Date(2022, 1, 18, 14, 15),
+    },
+    {
+        title: 'Upgrade Personal Computers',
+        startDate: new Date(2022, 1, 19, 15, 15),
+        endDate: new Date(2022, 1, 20, 20, 30),
+    },
+    {
+        title: 'Prepare 2015 Marketing Plan',
+        startDate: new Date(2022, 1, 20, 20, 0),
+        endDate: new Date(2022, 1, 20, 13, 30),
+    },
+    {
+        title: 'Brochure Design Review',
+        startDate: new Date(2022, 1, 20, 14, 10),
+        endDate: new Date(2022, 1, 20, 15, 30),
+    },
+];
+
+const ExternalViewSwitcher = ({ currentViewName, onChange }) => (
+    <ToggleButtonGroup
+        color="primary"
+        value={currentViewName}
+        exclusive
+        onChange={onChange}
+        sx={{ display: 'flex', justifyContent: 'flex-end', pb: 2 }}
+    >
+        <ToggleButton
+            sx={{
+                textTransform: 'none',
+                fontWeight: 400,
+                color: '#373F41',
+            }}
+            value="Day"
+        >
+            Day
+        </ToggleButton>
+        <ToggleButton
+            sx={{
+                textTransform: 'none',
+                fontWeight: 400,
+                color: '#373F41',
+            }}
+            value="Week"
+        >
+            Week
+        </ToggleButton>
+        <ToggleButton
+            sx={{
+                textTransform: 'none',
+                fontWeight: 400,
+                color: '#373F41',
+            }}
+            value="Month"
+        >
+            Month
+        </ToggleButton>
+    </ToggleButtonGroup>
+);
+
+export default class MainScheduler extends React.PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            data: appointments,
+            currentViewName: 'Month',
+        };
+
+        this.currentViewNameChange = (e) => {
+            this.setState({ currentViewName: e.target.value });
+        };
+    }
+
+    render() {
+        const { data, currentViewName } = this.state;
+
+        return (
+            <Paper elevation={3}>
+                <ExternalViewSwitcher currentViewName={currentViewName} onChange={this.currentViewNameChange} />
+                <Scheduler data={data} height={700}>
+                    <ViewState currentViewName={currentViewName} />
+                    <DayView startDayHour={8} endDayHour={18} />
+                    <WeekView startDayHour={8} endDayHour={18} />
+                    <MonthView />
+                    <Toolbar />
+                    <DateNavigator />
+                    <TodayButton />
+                    <Appointments />
+                </Scheduler>
+            </Paper>
+        );
+    }
+}

```

Comment:
Here you're using class components. You can try to translate this component in functional components

Reply:
Ok fixed!